src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -4,18 +4,27 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Diagnostics;
-using Microsoft.CodeAnalysis.Collections;
 using Microsoft.CodeAnalysis.CSharp.Symbols;
-using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.PooledObjects;
-using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal sealed partial class OverloadResolution
     {
         public void BinaryOperatorOverloadResolution(BinaryOperatorKind kind, BoundExpression left, BoundExpression right, BinaryOperatorOverloadResolutionResult result, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        {
+            // We can do a table lookup for well-known problems in overload resolution.
+            BinaryOperatorOverloadResolution_EasyOut(kind, left, right, result);
+            if (result.Results.Count > 0)
+            {
+                return;
+            }
+
+            BinaryOperatorOverloadResolution_NoEasyOut(kind, left, right, result, ref useSiteDiagnostics);
+        }
+
+        internal void BinaryOperatorOverloadResolution_EasyOut(BinaryOperatorKind kind, BoundExpression left, BoundExpression right, BinaryOperatorOverloadResolutionResult result)
         {
             Debug.Assert(left != null);
```

Comment:
Do we want to do these asserts in `BinaryOperatorOverloadResolution` as opposed to just the EasyOut method?

Reply:
I meant to include the asserts in both new methods. I'll add the asserts in the next PR.

---
In reply to: [355605110](https://github.com/dotnet/roslyn/pull/39157#discussion_r355605110) [](ancestors = 355605110)